cmd/posix.go

```diff
@@ -672,11 +672,15 @@ func (s *posix) Walk(volume, dirPath, marker string, recursive bool, leafFile st
 		return nil, err
 	}
 
-	ch = make(chan FileInfo)
+	// buffer channel matches the S3 ListObjects implementation
+	ch = make(chan FileInfo, maxObjectList)
```

Comment:
By the way, a similar change in Walk() in storage-rest-client.go could also be efficient

Reply:
Yeah that's not done on purpose because there can be many disks to avoid memory build up.